### PR TITLE
tkt-36473: feat(zsh): improve prompt and add useful bindkeys

### PR DESCRIPTION
**Pull request by william-gr**
Ticket:	#36473

### CODE REVIEW

File: src/freenas/root/.zshrc [hunk 1]
```diff
@@ -1,9 +1,14 @@
 
 HISTFILE=~/.zsh-histfile
 SAVEHIST=1000
+PROMPT="%n@%m# "
```
Comment (themylogin) [inline src/freenas/root/.zshrc]: Can we also include `pwd` then? Like `%20<...<%~%<<`

File: src/freenas/root/.zshrc [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
 
 setopt APPEND_HISTORY
 
+bindkey "^[[F" end-of-line
+bindkey "^[[H" beginning-of-line
+bindkey "^[[3~" delete-char
+
```
Comment (themylogin) [inline src/freenas/root/.zshrc]: I also suggest to include this:

```
# Alt+Left/Right
bindkey '\e[1;5D' backward-word
bindkey '\e[1;5C' forward-word
# Alt+Delete
bindkey '\e\e[3~' delete-word
# Ctrl+Backspace
bindkey '^H' backward-delete-word
```

Comment (themylogin) [reply src/freenas/root/.zshrc]: `set -o emacs` cancels it (but running `bindkey` again brings it back)

Emacs mode probably offers combinations for these actions so nevermind then

File: src/freenas/root/.zshrc (hunk 1 above)
Comment (themylogin) [reply src/freenas/root/.zshrc]: Cuts path from the start if it is too long. Probably `%40<...<%~%<<` would be a better width.

**Comment by william-gr** [comment]
@themylogin how is that now?

File: src/freenas/root/.zshrc [hunk 3]
```diff
@@ -1,9 +1,15 @@
...
 setopt APPEND_HISTORY
 
+bindkey "^[[F" end-of-line
+bindkey "^[[H" beginning-of-line
+bindkey "^[[3~" delete-char
+bindkey '^H' backward-delete-word
+
 # Enable the builtin emacs(1) command line editor in sh(1),
 # e.g. C-a -> beginning-of-line.
 set -o emacs
```
Comment (jhixson74) [inline src/freenas/root/.zshrc]: Is this necessary? Not everyone uses emacs or wants emacs key bindings :-/

Comment (william-gr) [reply src/freenas/root/.zshrc]: I dont use emacs either, but thats the default of zsh. We cant have both ways. Do you prefer we dont have any keys by default?

Comment (jhixson74) [reply src/freenas/root/.zshrc]: Disregard. I looked at default .zshrc. Of course key bindings are useful, I just didn't think we want to force anyone one way or the other, but it's easy enough to switch.

**Pull request merged**